.github/workflows/lint.yml

```diff
@@ -0,0 +1,50 @@
+name: Lint
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  run-linters:
+    name: Run linters
+    runs-on: ubuntu-latest
+    env:
+      NPM_VERSION: "8.3.0"
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14'
+          cache: 'npm'
+          cache-dependency-path: 'frontend/package-lock.json'
+
+      - name: Install dependencies
+        run: npm install --global npm@${{ env.NPM_VERSION }}
+
+      - name: "Install dependencies: workspace"
+        working-directory: frontend
+        run: npm install
+
+
+      - name: "Run ESlint: common"
+        working-directory: frontend/common
+        continue-on-error: true
+        run: |
+          npm run codegen
+          npm run lint
+
+      - name: "Run ESlint: admin"
+        working-directory: frontend/admin
+        continue-on-error: true
+        run: |
+          npm run codegen
+          npm run lint
+
+      - name: "Run ESlint: talentSearch"
+        working-directory: frontend/talentsearch
+        continue-on-error: true
+        run: |
+          npm run codegen
+          npm run lint
```

Comment:
Maybe this is worse for failures, but recently learned that this is possible for workspaces:

https://docs.npmjs.com/cli/v7/using-npm/workspaces#ignoring-missing-scripts

```suggestion
      - name: "Run ESlint: all workspaces"
        working-directory: frontend
        continue-on-error: true
        run: |
          npm run codegen --workspaces --if-present
          npm run lint --workspaces --if-present
```

Also, `continue-on-error`: does that raise an warning at all, or just skip it? I always use `if: ${{ always() }}` on subsequent steps (after one I think might fail, but that I was still to run, like more tests). I think `continue-on-error` will totally hide errors in PRs, instead of just moving past them in subsequent steps?

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error

> Prevents a job from failing when a step fails. Set to true to allow a job to pass when this step fails.

Reply:
Sorry, this is confusing. Happy to stick with how we were doing it, but just remove `continue-on-error` like so:

```suggestion
      - name: "Run ESlint: common"
        working-directory: frontend/common
        if: ${{ always() }}
        run: |
          npm run codegen
          npm run lint

      - name: "Run ESlint: admin"
        working-directory: frontend/admin
        if: ${{ always() }}
        run: |
          npm run codegen
          npm run lint

      - name: "Run ESlint: talentSearch"
        working-directory: frontend/talentsearch
        if: ${{ always() }}
        run: |
          npm run codegen
          npm run lint
```

Reply:
checkout this link :
https://github.com/GCTC-NTGC/gc-digital-talent/runs/5190508335?check_suite_focus=true 
Continue on error doesn't hide any errors. It just move on to next step without exiting the yml file and make the final status as "Success"

Reply:
sorry, when i say "hide errors" i don't mean that it hides them from logs when someone goes to look, but rather than it hides them in the status checks.

<img width="878" alt="Screen Shot 2022-02-15 at 11 44 52 AM" src="https://user-images.githubusercontent.com/305339/154108308-a0af38ff-fc34-4540-9654-243944c99ff1.png">

So if I'm understanding your link correctly, it shows "run linters" as green checkmark, even when some steps failed. And so in the PR, it will show "run linters" as passing, and no one would know to check the logs.

Am I understanding that correctly?

Reply:
But maybe you could help me understand: would we want final status to be success when there are errors?

Reply:
```suggestion
      - name: "Setup: all workspaces"
        working-directory: frontend
        run: npm run codegen -- workspaces --if-present

      - name: "Run ESlint: all workspaces"
        working-directory: frontend
        continue-on-error: true
        run: npm run lint
```